recipes/geos/all/conanfile.py

```diff
@@ -96,8 +93,14 @@ def generate(self):
         tc.variables["BUILD_GEOSOP"] = self.options.utils
         tc.generate()
 
+    def _patch_sources(self):
+        # Avoid setting CMAKE_BUILD_TYPE default when multi-config generators are used.
+        # https://github.com/libgeos/geos/pull/945
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                        "set(CMAKE_BUILD_TYPE ${DEFAULT_BUILD_TYPE})", "", strict=False)
```

Comment:
By having this not be strict, we could get into a situation where the patch is no longer needed but we don't realize. Any reason not to have it?

Reply:
Fixed.